Support serializing to the PKCS#12 file format.

boumenot opened the pull request.

paulmey commented on packer/builder/azure/pkcs12/pbkdf_test.go:
```diff
@@ -0,0 +1,20 @@
+	"testing"
+)
+
+var hit = false
```
not sure what this is for...

boumenot replied:
I don't know either, it was from the original code.  I'm removed it.

paulmey commented:
Can we remove everything that is not necessary for encoding? We should be using the x/crypto package for decoding.

boumenot commented:
I removed all of the top level decoding pieces.  The lower level decoding pieces are useful for unit testing because it's easier to assert that something round trips rather than it just encrypts.  I moved all decoding routines into the _test files to make the intent clearer.
